lib/TextDecoder-impl.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const { labelToName, isSupported, decode } = require("./whatwg-encoding");
+
+exports.implementation = class TextDecoderImpl {
+  constructor(globalObject, constructorArgs = []) {
+    const label = constructorArgs[0];
+    const options = constructorArgs[1] || {};
+
+    const encoding = labelToName(label);
+    if (!isSupported(encoding) || encoding === "replacement") {
+      throw new RangeError(`"${encoding}" is not a supported encoding name`);
+    }
+    this._encoding = encoding;
+    this._errorMode = options._fatal ? "fatal" : "replacement";
+    this._ignoreBOM = options._ignoreBOM || false;
+  }
+
+  get encoding() {
+    return String(this._encoding).toLowerCase();
+  }
+
+  get fatal() {
+    return this._errorMode === "fatal";
+  }
+
+  get ignoreBOM() {
+    return this._ignoreBOM;
+  }
+
+  decode(input, options = {}) {
+    if (options.steam === true) {
```

Comment:
```suggestion
    if (options.stream === true) {
```

You may want to look into https://github.com/ashtuchkin/iconv-lite/blob/master/lib/streams.js to see how iconv-lite streaming works.